MetaMask/metamask-extension
# Deprecate JS version of Box in favor of TS version

## Explanation
Currently there are 2 version of the `Box` component. The JS version and TS version. We want to prevent the usage of the JS version and promote the usage of the TS version. This PR deprecates the JS version and directs engineers on how to update their files to use the new TS version of the `Box` component. 

Even though I am a speaker and writer of the Queen's english I will concede to the US spelling and have also updated all instances of "favour" to "favor" which @legobeat has mentioned in a previous deprecation PR

* Fixes #19529 

## Screenshots/Screencaps

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/cc8fcae5-0add-4730-b0e9-e2f5529dd249

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/34178500-5d9c-4c4e-bb98-3c6bd48f1989

![Screenshot 2023-06-08 at 1 01 29 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/abce8d9b-f8a3-40dd-908b-fbdb1a6d5739)
![Screenshot 2023-06-08 at 12 26 08 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/7e4d66f8-88b2-4db1-8203-1d9a1898d030)
![Screenshot 2023-06-08 at 12 29 49 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/01eea2d2-423d-4f3a-a0b5-636e2db7b5cc)
<img width="1438" alt="Screenshot 2023-06-08 at 12 56 13 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/4bf400cd-5610-4a26-9116-62c139090009">

## Manual Testing Steps

- Pull this branch
- Search for the js version of the Box component `<Box`
- See that a deprecation message has been added

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
